File: frontend/src/app/models/expense.ts
export type Category = 'Food' | 'Transport' | 'Housing' | 'Health' | 'Entertainment' | 'Other';
export type PaymentMethod = 'Cash' | 'CreditCard' | 'DebitCard' | 'BankTransfer' | 'EWallet';

export interface Expense {
  id: number;
  title: string;
  amount: number;
  category: Category;
  date: string;
  notes?: string;
  paymentMethod?: PaymentMethod;
}

File: backend/Models/Expense.cs
namespace ExpenseTracker.API.Models;

public class Expense
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public Category Category { get; set; }
    public DateOnly Date { get; set; }
    public string? Notes { get; set; }
    public PaymentMethod? PaymentMethod { get; set; }
}
